ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -1007,12 +1007,16 @@ void checkAndMaybeComplete() {
       if (progressTracker.isDone()) {
         if (progressTracker.hasSucceeded() && !retainChunkExceptionOnSuccess) {
           chunkException = null;
-        } else if (chunkOperationTracker.maybeFailedDueToOfflineReplicas()) {
-          chunkException =
-              buildChunkException("Get Chunk failed because of offline replicas", RouterErrorCode.AmbryUnavailable);
-        } else if (chunkOperationTracker.hasFailedOnNotFound()){
-          chunkException =
-              buildChunkException("Get Chunk failed because of BlobNotFound", RouterErrorCode.BlobDoesNotExist);
+        } else {
+          if (chunkOperationTracker.hasFailedOnNotFound()) {
+            // If operation failed due to blob not being present, send 404.
+            chunkException =
+                buildChunkException("Get Chunk failed because of BlobNotFound", RouterErrorCode.BlobDoesNotExist);
+          } else if (chunkOperationTracker.hasFailedOnUnavailability()) {
+            // If some originating DC replicas are down, and we can't conclude if blob is missing, send 503.
+            chunkException =
```

Comment:
hasFailedOnUnavailability only means there is some replica unavailable, not necessary mean we fail on unavailable.
Seems chunkException is overwriting the previous exception. It'll overwrite "deleted" or "expired" status.